src/InstallWizard.js

```diff
@@ -178,7 +241,22 @@ class InstallWizard extends Component {
     props.saveModel = this.saveModel;
     props.loadModel = this.loadModel;
 
-    return React.createElement(this.props.pages[this.state.currentStep].component, props);
+    //update
+    if (this.state.currentMenuName) {
+      props.startUpdateProcess = this.startUpdate;
+      props.closeUpdateProcess = this.closeUpdate;
+      props.cancelUpdateProcess = this.cancelUpdate;
+    }
+
+    //don't have any progress steps
+    if (this.state.currentStep === undefined) {
+      return React.createElement(this.props.menuComponent, props);
+    }
+    else {
+      //if it is update, it will use dynamically assembled this.state.pages when startUpdate is
+      //triggered, if it is install, it will use the static this.props.pages
```

Comment:
if what is update? is this basically splitting between "install" and "configure" modes?

Reply:
I will use a const to indicate if it is update to make it a little clearer